src/library_sockfs.js

```diff
@@ -303,15 +303,21 @@ mergeInto(LibraryManager.library, {
         };
 
         function handleMessage(data) {
-          assert(typeof data !== 'string' && data.byteLength !== undefined);  // must receive an ArrayBuffer
-
-          // An empty ArrayBuffer will emit a pseudo disconnect event
-          // as recv/recvmsg will return zero which indicates that a socket
-          // has performed a shutdown although the connection has not been disconnected yet.
-          if (data.byteLength == 0) {
-            return;
+          if (typeof data === 'string') {
+            assert(typeof TextEncoder !== undefined); // required for string encoding
```

Comment:
I think this assertion is not necessary, as it will throw on the next line anyhow, if that is not present.

Reply:
Ok, the line removed and the changes rebased on the `incoming` branch. Done.